TourMate/TourMate/Frontend/User/Controllers/FirebaseUserController.swift

```diff
@@ -46,4 +46,27 @@ struct FirebaseUserController: UserController {
         return (userAdapter.toUser(adaptedUser: adaptedUser), errorMessage)
     }
 
+    func getUser(email: String) async -> (User?, String) {
+        guard Auth.auth().currentUser != nil else {
+            return (nil, Constants.messageUserNotLoggedIn)
+        }
+
+        let (adaptedUsers, errorMessage) = await firebasePersistenceManager.fetchItems(field: "email", isEqualTo: email)
+
+        guard errorMessage.isEmpty else {
+            return (nil, errorMessage)
+        }
+
+        // should not have more than 1 user with the same email
+        guard adaptedUsers.count <= 1 else {
+            preconditionFailure()
+        }
```

Comment:
Although this should not happen, `preconditionFailure()` would kill the app so I'm thinking we should do something safer?
```suggestion
        guard adaptedUsers.count <= 1 else {
            assertionFailure()
            return (nil, "More than one user found")
        }
```

Reply:
Ah ok, I was following `getUser()` to be consistent. Lemme update both functions.

Reply:
> Do you think we should include a list view of all the attending users in `TripVIew`?

Okie, I'll add it in as well